File: server/models/attendance.js
'use strict';
const {
  Model
} = require('sequelize');

module.exports = (sequelize, DataTypes) => {
  class Attendance extends Model {
    static associate(models) {
      Attendance.belongsTo(models.Employee, {
        foreignKey: 'employeeId',
        as: 'employee'
      });
    }
  }

  Attendance.init({
    employeeId: DataTypes.INTEGER,
    date: DataTypes.DATEONLY,
    timeIn: DataTypes.TIME,
    timeOut: DataTypes.TIME,
    break1In: DataTypes.TIME,
    break1Out: DataTypes.TIME,
    lunchIn: DataTypes.TIME,
    lunchOut: DataTypes.TIME,
    break2In: DataTypes.TIME,
    break2Out: DataTypes.TIME,
    break3In: DataTypes.TIME,
    break3Out: DataTypes.TIME,
    status: DataTypes.STRING,
    lateMinutes: DataTypes.INTEGER
  }, {
    sequelize,
    modelName: 'Attendance',
  });

  return Attendance;
};

File: android_app/lib/models/attendance.dart
class Attendance {
  final int id;
  final int employeeId;
  final String date;
  final String timeIn;
  final String timeOut;
  final String status;
  final int lateMinutes;

  Attendance({
    required this.id,
    required this.employeeId,
    required this.date,
    required this.timeIn,
    required this.timeOut,
    required this.status,
    required this.lateMinutes,
  });

  factory Attendance.fromJson(Map<String, dynamic> json) {
    return Attendance(
      id: json['id'],
      employeeId: json['employeeId'],
      date: json['date'],
      timeIn: json['timeIn'],
      timeOut: json['timeOut'],
      status: json['status'],
      lateMinutes: json['lateMinutes'],
    );
  }
}
